include/solvers/multiphysics_interface.h

```diff
@@ -764,15 +849,25 @@ class MultiphysicsInterface
    * NB : only implemented for VOF for now
    */
   virtual void
-  compute_kelly(dealii::Vector<float> &estimated_error_per_cell)
+  compute_kelly(Parameters::MeshAdaptation::Variable mesh_adaptation_variable,
+                dealii::Vector<float> &              estimated_error_per_cell)
   {
-    for (auto &iphys : physics)
+    if (mesh_adaptation_variable ==
+        Parameters::MeshAdaptation::Variable::velocity_temperature)
       {
-        iphys.second->compute_kelly(estimated_error_per_cell);
+        physics[PhysicsID::heat_transfer]->compute_kelly(
+          estimated_error_per_cell);
       }
-    for (auto &iphys : block_physics)
+    else
       {
-        iphys.second->compute_kelly(estimated_error_per_cell);
+        for (auto &iphys : physics)
+          {
+            iphys.second->compute_kelly(estimated_error_per_cell);
+          }
```

Comment:
Can you explain what brought upon this change? I don't understand why this mechanism is needed?
If you want to combine the error per cell of the temperature and the velocity, you can already do it with the Vector<float> that is calculated here?

Reply:
this is also not used anymore.